### PR TITLE
{geo}[gompi/2021b] BiasAdjustCXX v1.9.1

### PR DESCRIPTION
**Pull request by TopRichard**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by jfgrimm** [comment]
@boegelbot please test @ generoso

**Comment by jfgrimm** [comment]
```
== 2023-12-02 15:10:50,746 easyblock.py:4277 WARNING build failed (first 300 chars): Failed to determine CMake version from output of 'cmake --version': /bin/bash: cmake: command not found
```

File: easybuild/easyconfigs/b/BiasAdjustCXX/BiasAdjustCXX-1.9.1-foss-2021b.eb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+source_urls = ['https://github.com/btschwertfeger/BiasAdjustCXX/archive/refs/tags/']
+sources = ['v%(version)s.tar.gz']
+checksums = ['9f2988d0b1676c57037cdb5cbb4e313b31c4fc08a1ea8a55ea5eccc5a2bc5861']
+
+dependencies = [
+    ('netCDF', '4.8.1'),
+    ('CDO', '2.0.5'),
+    ('NCO', '5.0.3'),
+]
```
Comment (jfgrimm) [inline easybuild/easyconfigs/b/BiasAdjustCXX/BiasAdjustCXX-1.9.1-foss-2021b.eb:23]: you're missing a build dependency on `CMake` here
```suggestion
builddependencies = [
    ('CMake', '3.22.1'),
]

dependencies = [
    ('netCDF', '4.8.1'),
    ('CDO', '2.0.5'),
    ('NCO', '5.0.3'),
]
```

**Review by jfgrimm — CHANGES_REQUESTED** [review]

**Comment by jfgrimm** [comment]
@boegelbot please test @ generoso

**Comment by jfgrimm** [comment]
@TopRichard I've pushed a change that renames the easyconfig to reflect the toolchain change

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by smoors** [comment]
>CMakeFiles/BiasAdjustCXX.dir/Manager.cxx.o:Manager.cxx:function Manager::adjust_3d(std::vector<std::vector<std::vector<float, std::allocator<float> >, std::allocator<std::vector<float, std::allocator<float> > > >, std::allocator<std::vector<std::vector<float, std::allocator<float> >, std::allocator<std::vector<float, std::allocator<float> > > > > >&): error: undefined reference to 'pthread_create'

File: easybuild/easyconfigs/b/BiasAdjustCXX/BiasAdjustCXX-1.9.1-gompi-2021b.eb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    ('CMake', '3.22.1'),
+]
+
+dependencies = [
+    ('netCDF', '4.8.1'),
+    ('netCDF-C++4', '4.3.1'),
+    ('CDO', '2.0.5'),
+]
+
+preconfigopts = 'CXXFLAGS="$CXXFLAGS -lnetcdf " '
```
Comment (smoors) [inline easybuild/easyconfigs/b/BiasAdjustCXX/BiasAdjustCXX-1.9.1-gompi-2021b.eb:29]: you probably have to add -lpthread here:
```suggestion
preconfigopts = 'CXXFLAGS="$CXXFLAGS -lnetcdf -lpthread" '
```

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by smoors** [comment]
@boegelbot please test @ jsc-zen2

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @TopRichard!

**Pull request merged**